### ISSUE TITLE
⌨ Pool candidate, view application link lacking context

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#7105]
## ⌨ Accessibility Issue

Currently, the "view application" link on the pool candidates page does not provide enough context.

### WCAG 2.1 Criteria

[2.4.4 Link Purpose (in context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)

## 🦋 Expected Behaviour

Each link should provide enough context that its purpose is clear to a user in isolation.

## 📋 Steps to Reproduce

1. Navigate to the pool candidates table `/admin/pools/{poolId}/pool-candidates`
2. Open screen reader
3. Note the "view application" links are all identical while navigating to different locations.

## 📸 Screenshot

<img width="1512" alt="Screenshot 2023-06-27 at 11 58 16 AM" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/6d0c17ad-68f4-48f7-a659-aa398237059b">

## 🙋‍♀️ Proposed Solution

Add the users name to the link as either visually hidden text or an `aria-label`.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] View application links provide context
- [ ] Context is announced to ATs
